### DISCUSSION TITLE
Multiple Output Text Fields

### DISCUSSION THREAD

**Discussion by psalm119** [grantwinney/SurvivingWinForms#2] [Ideas]
Hi there, this project is very helpful, thank you! :)
Is it possible to have additional output text fields in the [Separate Thread example](https://github.com/grantwinney/Surviving-WinForms/blob/master/Threading/AsyncAwait/AsyncAwait/frmResponsiveModal.cs#L28)?

For example, in the [PourCoffee()](https://github.com/grantwinney/Surviving-WinForms/blob/master/Threading/AsyncAwait/AsyncAwait/Breakfast.cs#L30) method, if I add an HttpClient GetAsync() request to retrieve some data from a web API, I'd like the 3 string property values in the returned JSON object to populate into 3 text fields:
`txtSeparateThread.AppendText(strReturnValue1);`
`txt2SeparateThread.AppendText(strReturnValue2);`
`txt3SeparateThread.AppendText(strReturnValue3);`

I haven't been able to figure out how to expand the functionality with the `readonly Action<string> write;` and the `override void SendMessage(string text)`.

Kind regards.

Ben Williams

**Comment by grantwinney** [comment]
Hi, glad to hear it Ben. I don't have time right this moment to look at it, but I'll take a look over the next few days. :)

**Reply by psalm119** [reply]
Hi Grant, as it turns out, I was able to simply make use of the existing txtSeparateThread textbox as a temporary "placeholder" for my returned JSON data, then just parse _that_ into a model immediately after running the SingleThread operation.
`JsonConvert.DeserializeObject<MyResultModel>(txtJson).`

I'm certainly grateful to the good LORD for all the wisdom and patience HE gives us.

Best regards.

**Reply by grantwinney** [reply]
Glad to hear you found a solution that works for you! :)
